src/hotspot/share/opto/macro.cpp

```diff
@@ -853,6 +853,12 @@ SafePointScalarObjectNode* PhaseMacroExpand::create_scalarized_object_descriptio
   sobj->init_req(0, C->root());
   transform_later(sobj);
 
+  if (iklass && iklass->is_inlinetype()) {
+    // Value object has two additional inputs before the non-static fields
+    sfpt->add_req(_igvn.intcon(1));
+    sfpt->add_req(_igvn.intcon(alloc->_larval ? 1 : 0));
+  }
+
```

Comment:
@TobiHartmann , could you please help take a look at this change? I thought this should also fix at `lworld` branch, but currently I cannot reproduce it with a smaller case. Thanks so much!